ui/controls.js

```diff
@@ -223,6 +223,18 @@ shaka.ui.Controls = class extends shaka.util.FakeEventTarget {
     this.eventManager_.listen(this.player_, 'texttrackvisibility', () => {
       this.computeShakaTextContainerSize_();
     });
+
+    this.eventManager_.listen(this.player_, 'unloading', () => {
+      if (this.isFullScreenEnabled()) {
+        this.exitFullScreen_();
+      }
+      if (this.isPiPEnabled()) {
+        this.togglePiP();
+      }
+      if (this.config_.setupMediaSession && navigator.mediaSession) {
+        navigator.mediaSession.metadata = null;
+      }
```

Comment:
shaka-player should only clear the media session metadata if it was the one that set it. Checking if the config value is enabled isn't enough, as it is possible to have that config enable so that the controls work and use media that doesn't have embedded metadata and instead set it in the app code.

Reply:
I'm going to discard this code for the moment